plugins/polio/api/vaccines/supply_chain.py

```diff
@@ -40,8 +40,11 @@ def validate_rounds_and_campaign(data, current_user=None):
         raise forms.ValidationError("A campaign must be attached.")
 
     try:
-        new_campaign = Campaign.objects.get(obr_name=campaign_obr_name)
-        data["campaign"] = new_campaign
+        if isinstance(campaign_obr_name, Campaign):
+            new_campaign = campaign_obr_name
```

Comment:
Not blocking but reading this is a bit misleading.
It looks just like `campaign_obr_name` is a string while in fact it's a `Campaign` instance.
I had to go back the chain up to the model to understand.